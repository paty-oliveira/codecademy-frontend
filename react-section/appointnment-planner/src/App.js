import React, { useState } from "react";
import { Switch, Route, Redirect, NavLink } from "react-router-dom";

import { AppointmentsPage } from "./containers/appointmentsPage/AppointmentsPage";
import { ContactsPage } from "./containers/contactsPage/ContactsPage";

function App() {

  const [contacts, setContacts] = useState([]);
  const [appointments, setAppointments] = useState([]);

  const ROUTES = {
    CONTACTS: "/contacts",
    APPOINTMENTS: "/appointments",
  };

  const addContact = (name, phoneNumber, email) => {
    const newContact = {name: name, phoneNumber: phoneNumber, email: email};

    setContacts((previousContact) => {
      return [...previousContact, newContact]
    });
  };

  const addAppointment = (title, contact, date, time) => {
    const newAppointment = {title: title, contact: contact, date: date, time: time};

    setAppointments((previousAppointment) => {
      return [...previousAppointment, newAppointment]
    });
  };


  return (
    <>
      <nav>
        <NavLink to={ROUTES.CONTACTS} activeClassName="active">
          Contacts
        </NavLink>
        <NavLink to={ROUTES.APPOINTMENTS} activeClassName="active">
          Appointments
        </NavLink>
      </nav>
      <main>
        <Switch>
          <Route exact path="/">
            <Redirect to={ROUTES.CONTACTS} />
          </Route>
          <Route path={ROUTES.CONTACTS}>
            <ContactsPage contacts={contacts} addContact={addContact} />
          </Route>
          <Route path={ROUTES.APPOINTMENTS}>
            <AppointmentsPage contacts={contacts} appointments={appointments} addAppointment={addAppointment} />
          </Route>
        </Switch>
      </main>
    </>
  );
}

export default App;
