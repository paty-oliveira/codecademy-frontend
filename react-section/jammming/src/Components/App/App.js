import "./App.css";
import React from "react";
import { SearchBar } from "../SearchBar/SearchBar";
import { SearchResults } from "../SearchResults/SearchResults";
import { Playlist } from "../Playlist/Playlist";

class App extends React.Component {
  constructor(props){
    super(props);
    this.state = {
        searchResults: [
            {name: "Tiny Dancer", artist: "Elton John", album: "Madman Across The Water", id: 1},
            {name: "Tiny Dancer", artist: "Tim McGraw", album: "Love Story", id: 2},
            {name: "Tiny Dancer", artist: "The White Raven", album: "Tiny Dancer", id: 3}
        ],
        playlistName: "Playlist by Patricia",
        playlistTracks: [
            {name: "Stronger", artist: "Britney Spears", album: "Oops!... I Did it Again", id: 4},
            {name: "So Emotional", artist: "Whitney Houston", album: "Whitney", id: 5},
            {name: "It's Not Right But It's Okay", artist: "Whitney Houston", album: "My Love is Your Love", id: 6}
        ]
    };
    this.addTrack = this.addTrack.bind(this);
    this.removeTrack = this.removeTrack.bind(this);
    this.updatePlaylistName = this.updatePlaylistName.bind(this);
    this.savePlaylist = this.savePlaylist.bind(this);
	this.search = this.search.bind(this);
}

  addTrack(track) {
    let tracks = this.state.playlistTracks;
    if (!tracks.find(savedTrack => savedTrack.id === track.id)) {
        tracks.push(track);
    }

    this.setState({ playlistTracks: tracks });
  }

  removeTrack(track) {
    let tracks = this.state.playlistTracks;
    tracks = tracks.filter(currentTrack => currentTrack.id !== track.id);

    this.setState({ playlistTracks: tracks} );
  }

  updatePlaylistName(newPlaylistName) {
    this.setState({ playlistName: newPlaylistName });
  }

  savePlaylist() {
    const tracksUris = this.state.playlistTracks.map(track => track.uri);
  }

  search(searchItem) {
	console.log(searchItem);
  }

  render() {
    return (
      <div>
        <h1>Ja<span className="highlight">mmm</span>ing</h1>
        <div className="App">
          <SearchBar onSearch={this.search} />
          <div className="App-playlist">
            <SearchResults
                searchResults={this.state.searchResults}
                onAdd={this.addTrack} />
            <Playlist
                playlistName={this.state.playlistName}
                playlistTracks={this.state.playlistTracks}
                onRemove={this.removeTrack}
                onNameChange={this.updatePlaylistName}
                onSave={this.savePlaylist} />
          </div>
        </div>
      </div>
    )
  }
}


export default App;
