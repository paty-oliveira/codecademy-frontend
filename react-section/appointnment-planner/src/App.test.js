import React from "react";
import { shallow, mount } from 'enzyme';

import App from "./App";

describe('App', () => {
    const contactsProps = [{
        name: "patricia",
        phoneNumber: "910025679",
        email: "example@xpto.com"
    }]

});
    it('should render correctly', () => {
        const appComponent = shallow(<App />);

        expect(appComponent).toMatchSnapshot();
    });
;

