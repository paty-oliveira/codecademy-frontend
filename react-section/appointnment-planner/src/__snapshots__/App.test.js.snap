// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render correctly 1`] = `ShallowWrapper {}`;

exports[`should render correctly 1`] = `ShallowWrapper {}`;
